pkg/clients/device/device.go

```diff
@@ -45,7 +45,24 @@ func CreateFromDevice(d *v1alpha1.Device) *packngo.DeviceCreateRequest {
 	}
 }
 
-// NeedsUpdate ... TODO
+// NeedsUpdate returns true if the supplied Kubernetes resource differs from the
+// supplied Packet resource. It considers only fields that can be modified in
+// place without deleting and recreating the instance.
 func NeedsUpdate(d *v1alpha1.Device, p *packngo.Device) bool {
+	if d.Spec.Hostname != p.Hostname {
+		return true
+	}
+	if d.Spec.Locked != &p.Locked {
+		return true
+	}
 	return false
 }
+
+// NewUpdateDeviceRequest creates a request to update an instance suitable for
+// use with the Packet API.
+func NewUpdateDeviceRequest(d *v1alpha1.Device, p *packngo.Device) *packngo.DeviceUpdateRequest {
+	return &packngo.DeviceUpdateRequest{
```

Comment:
Similar to comment above, should we include everything in [`DeviceUpdateRequest`](https://github.com/packethost/packngo/blob/6292a62434aeffca0d2cca5c2079349c2fc64650/devices.go#L166)?

Reply:
As I extended `DeviceProperties` now we can do the check. Although I think if only one is changed then we can proceed to update all. The operation would be idempotent.

Reply:
Right, I agree we can update all, but here we are only choosing to update `Hostname` and `Locked`, while our `spec` includes many other fields which can also be included in a `DeviceUpdateRequest`.

Reply:
I'll let you know once I have it updated.